extra/modules/greenbids-real-time-data/src/main/java/org/prebid/server/hooks/modules/greenbids/real/time/data/v1/GreenbidsRealTimeDataProcessedAuctionRequestHook.java

```diff
@@ -0,0 +1,202 @@
+                .recover(throwable -> Future.succeededFuture(toInvocationResult(
+                        bidRequest, null, InvocationAction.no_action)));
+    }
+
+    private Partner parseBidRequestExt(BidRequest bidRequest) {
+        return Optional.ofNullable(bidRequest)
+                .map(BidRequest::getExt)
+                .map(ExtRequest::getPrebid)
+                .map(ExtRequestPrebid::getAnalytics)
+                .filter(this::isNotEmptyObjectNode)
+                .map(analytics -> (ObjectNode) analytics.get(BID_REQUEST_ANALYTICS_EXTENSION_NAME))
+                .map(this::toPartner)
+                .orElse(null);
+    }
+
+    private boolean isNotEmptyObjectNode(JsonNode analytics) {
+        return analytics != null && analytics.isObject() && !analytics.isEmpty();
+    }
+
+    private Partner toPartner(ObjectNode adapterNode) {
+        try {
+            return mapper.treeToValue(adapterNode, Partner.class);
+        } catch (JsonProcessingException e) {
+            return null;
+        }
+    }
+
+    private Future<InvocationResult<AuctionRequestPayload>> toInvocationResult(
+            BidRequest bidRequest,
+            Partner partner,
+            OnnxModelRunner onnxModelRunner,
+            Double threshold) {
+
+        final Map<String, Map<String, Boolean>> impsBiddersFilterMap;
+        try {
+            final List<ThrottlingMessage> throttlingMessages = greenbidsInferenceDataService
+                    .extractThrottlingMessagesFromBidRequest(bidRequest);
+
+            impsBiddersFilterMap = filterService.filterBidders(
+                    onnxModelRunner,
+                    throttlingMessages,
+                    threshold);
+        } catch (PreBidException e) {
+            return Future.succeededFuture(toInvocationResult(
+                    bidRequest, null, InvocationAction.no_action));
+        }
+
+        final GreenbidsInvocationResult greenbidsInvocationResult = greenbidsInvocationService
+                .createGreenbidsInvocationResult(partner, bidRequest, impsBiddersFilterMap);
+
+        return Future.succeededFuture(toInvocationResult(
+                greenbidsInvocationResult.getUpdatedBidRequest(),
+                greenbidsInvocationResult.getAnalyticsResult(),
+                greenbidsInvocationResult.getInvocationAction()));
+    }
+
+    private InvocationResult<AuctionRequestPayload> toInvocationResult(
+            BidRequest bidRequest,
+            AnalyticsResult analyticsResult,
+            InvocationAction action) {
+
+        final List<AnalyticsResult> analyticsResults = (analyticsResult != null)
+                ? Collections.singletonList(analyticsResult)
+                : Collections.emptyList();
+
+        return InvocationResultImpl
+                .<AuctionRequestPayload>builder()
+                .status(InvocationStatus.success)
+                .action(action)
+                .payloadUpdate(payload -> AuctionRequestPayloadImpl.of(bidRequest))
```

Comment:
No need for payloadUpdate if action isn't update.

Reply:
Modified returned InvocationResult in case if action is not update

```java
InvocationResultImpl
                    .<AuctionRequestPayload>builder()
                    .status(InvocationStatus.success)
                    .action(action)
                    .analyticsTags(toAnalyticsTags(analyticsResults))
                    .build();
```